module/apmawssdkgo/sqs_test.go

```diff
@@ -0,0 +1,221 @@
+					AttributeNames: aws.StringSlice([]string{
+						"SentTimestamp",
+					}),
+					MaxNumberOfMessages: aws.Int64(1),
+					MessageAttributeNames: aws.StringSlice([]string{
+						"All",
+					}),
+					WaitTimeSeconds: aws.Int64(1),
+				})
+			},
+		},
+		{
+			name:            "SQS SEND to OtherQueue",
+			action:          "send",
+			resource:        "sqs/OtherQueue",
+			queueURL:        "https://sqs.testing.invalid/123456789012/OtherQueue",
+			hasTraceContext: true,
+			fn: func(ctx context.Context, svc *sqs.SQS, queueURL string) {
+				svc.SendMessageWithContext(ctx, &sqs.SendMessageInput{
+					QueueUrl: &queueURL,
+					MessageAttributes: map[string]*sqs.MessageAttributeValue{
+						"attr": &sqs.MessageAttributeValue{
+							DataType:    aws.String("String"),
+							StringValue: aws.String("string attr"),
+						},
+					},
+					MessageBody: aws.String("msg body"),
+				})
+			},
+		},
+		{
+			name:            "SQS SEND_BATCH to OtherQueue",
+			action:          "send_batch",
+			resource:        "sqs/OtherQueue",
+			queueURL:        "https://sqs.testing.invalid/123456789012/OtherQueue",
+			hasTraceContext: true,
+			fn: func(ctx context.Context, svc *sqs.SQS, queueURL string) {
+				svc.SendMessageBatchWithContext(ctx, &sqs.SendMessageBatchInput{
+					QueueUrl: &queueURL,
+					Entries: []*sqs.SendMessageBatchRequestEntry{
+						{
+							Id: aws.String("1"),
+							MessageAttributes: map[string]*sqs.MessageAttributeValue{
+								"attr": &sqs.MessageAttributeValue{
+									DataType:    aws.String("String"),
+									StringValue: aws.String("string attr"),
+								},
+							},
+							MessageBody: aws.String("msg body"),
+						},
+					},
+				})
+			},
+		},
+		{
+			name:     "SQS DELETE from ThatQueue",
+			action:   "delete",
+			resource: "sqs/ThatQueue",
+			queueURL: "https://sqs.testing.invalid/123456789012/ThatQueue",
+			fn: func(ctx context.Context, svc *sqs.SQS, queueURL string) {
+				svc.DeleteMessageWithContext(ctx, &sqs.DeleteMessageInput{
+					QueueUrl:      &queueURL,
+					ReceiptHandle: aws.String("receiptHandle"),
+				})
+			},
+		},
+		{
+			ignored: true,
+			fn: func(ctx context.Context, svc *sqs.SQS, _ string) {
+				svc.CreateQueueWithContext(ctx, &sqs.CreateQueueInput{
+					QueueName: aws.String("SQS_QUEUE_NAME"),
+				})
+			},
+		},
+	} {
+		ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if tc.hasError {
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+		}))
+		defer ts.Close()
+
+		region := "us-west-2"
+		cfg := aws.NewConfig().
+			WithEndpoint(ts.URL).
+			WithRegion(region).
+			WithDisableSSL(true).
+			WithCredentials(credentials.AnonymousCredentials)
+
+		session := session.Must(session.NewSession(cfg))
+		wrapped := WrapSession(session)
+		if tc.hasTraceContext {
+			wrapped.Handlers.Build.PushBackNamed(request.NamedHandler{
+				Name: "spy_message_attrs_added",
+				Fn:   testTracingAttributes(t),
```

Comment:
Took some inspiration from [the ruby client test](https://github.com/elastic/apm-agent-ruby/blob/4a3155ecebb7a189388f734d6a7478a05e80bc2f/spec/elastic_apm/spies/sqs_spec.rb#L68) and added another handler to inspect that the params were correctly modified.

Reply:
nice!